### ISSUE TITLE
Question about changing word level transformer to character level

### ISSUE THREAD

**Issue by Shamepoo** [ShuvenduBikash/transformer_spelling_corrector#1]
Hi,

I'm trying to modify a word level transformer to a character one to do some spelling correction. The source code is from:https://pytorch.org/tutorials/beginner/translation_transformer.html
I shared the embedding weight and projection weight. But my result was terrible, even when I put src and tgt same as gold text. The greedy decoder cannot even form a correct English word. I noticed that my val loss would not reduce after several epochs' training.
![image](https://user-images.githubusercontent.com/49515860/128374620-69c46dc8-73b1-427e-8fca-0c465bacdc59.png)
(This is a model with much less layers and heads)
I tried a lot of parameter combinations but none of them worked.
Your model works very well when I run the training process. In first 3000 steps, the results is significant.
May I ask is there any key modification should be done in character level transformer? (The vocabulary size of mine is 80+, including caps, not sure does this matter)

Thank you :)
